R/metadata_to_coldata.R

```diff
@@ -0,0 +1,69 @@
+#' Add sample metadata to colData of SCE object
+#'
+#' @param sce SingleCellExperiment object to add sample metadata to colData.
+#'   Must contain the `batch_column` as a column in `colData`.
+#' @param join_columns A character value giving the names of the columns in colData
+#'  to use for joining with the `sample_metadata`. Default is `library_id`
+#'
+#' @return SingleCellExperiment object with the contents of `metadata(sce)$sample_metadata`
+#'   added as new columns to the `colData(sce)`.
+#' @import SingleCellExperiment
+#'
+#' @export
+
+metadata_to_coldata <- function(sce,
+                                join_columns = "library_id") {
+
+  # make sure input is sce
+  if (!is(sce, "SingleCellExperiment")) {
+    stop("`sce` must be a SingleCellExperiment object.")
+  }
+
+  # check that sample_metadata present in metadata of sce object
+  # if it's not present, just skip everything and return the original object
+  metadata_list <- metadata(sce)
+  if(!"sample_metadata" %in% names(metadata_list)){
+    warning("No `sample_metadata` in SCE object to add.")
+  } else{
```

Comment:
Rather than indenting/nesting everything below, we can just short circuit the function here:
```suggestion
  if(!"sample_metadata" %in% names(metadata_list)){
    warning("No `sample_metadata` in SCE object to add.")
    return(sce)
  }
```

Since this is more parallel what happens with `stop()` calls, I find this to be more intuitive and easier to track than nesting the rest in an `else{}` block.